### ISSUE TITLE
Segfault with big values

### ISSUE THREAD

**Issue by guimard** [kazeburo/WWW-Form-UrlEncoded-XS#3]
Hello,

this simple script produce a segfault:
```
use WWW::Form::UrlEncoded::XS qw(build_urlencoded);
print build_urlencoded( 'Test' => ( 'x' x 50003 ) );
```
The limit seems exactly 50003 _(works with 50002)_. I found this bug when trying to encode a SAML POST form for [LemonLDAP::NG](https://lemonldap-ng.org) tests.

Cheers,
Xavier

**Comment by guimard** [comment]
Platform: Debian/testing

Version: 0.24

**Comment by kazeburo** [comment]
segfault was occurred in my Mac too.

```
% perl -e "use WWW::Form::UrlEncoded::XS qw(build_urlencoded);
dquote> print build_urlencoded( 'Test' => ( 'x' x 50003 ) );"
perl(16589,0x7fff989c83c0) malloc: *** error for object 0x7fa83de13458: incorrect checksum for freed object - object was probably modified after being freed.
*** set a breakpoint in malloc_error_break to debug
zsh: abort      perl -e 
```

**Comment by kazeburo** [comment]
Released 0.25.
Thank you for reporting!

**Comment by guimard** [comment]
Tested and [uploaded to Debian](https://salsa.debian.org/perl-team/modules/packages/libwww-form-urlencoded-xs-perl). Thanks !
